### PR TITLE
feat(DWS): add a new resource to manage snapshot

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
Running tool: /usr/local/go/bin/go test -timeout 9000000000s -run ^TestAccDwsSnapshot_basic$ github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dws -v

=== RUN   TestAccDwsSnapshot_basic
=== PAUSE TestAccDwsSnapshot_basic
=== CONT  TestAccDwsSnapshot_basic
--- PASS: TestAccDwsSnapshot_basic (1680.78s)
PASS
ok  	github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dws	1680.855s
```

### CODE REVIEW

File: huaweicloud/services/acceptance/dws/resource_huaweicloud_dws_snapshot_test.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+		},
+	})
+}
+
+func testDwsSnapshot_basic(name string) string {
+	return fmt.Sprintf(`
+%s
+
+resource "huaweicloud_dws_snapshot" "test" {
+  name = "%s"
```
Comment (houpeng80) [inline huaweicloud/services/acceptance/dws/resource_huaweicloud_dws_snapshot_test.go]:
```suggestion
  name        = "%s"
```

File: huaweicloud/services/dws/resource_huaweicloud_dws_snapshot.go [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+func ResourceDwsSnapshot() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceDwsSnapshotCreate,
+		ReadContext:   resourceDwsSnapshotRead,
+		DeleteContext: resourceDwsSnapshotDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/dws/resource_huaweicloud_dws_snapshot.go:40]: missing timeouts block

File: huaweicloud/services/acceptance/dws/resource_huaweicloud_dws_snapshot_test.go (hunk 1 above)
Comment (hmbig2) [reply huaweicloud/services/acceptance/dws/resource_huaweicloud_dws_snapshot_test.go]: done

File: huaweicloud/services/dws/resource_huaweicloud_dws_snapshot.go (hunk 2 above)
Comment (hmbig2) [reply huaweicloud/services/dws/resource_huaweicloud_dws_snapshot.go:40]: done

File: huaweicloud/services/dws/resource_huaweicloud_dws_snapshot.go [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+				Description: `ID of the cluster for which you want to create a snapshot.`,
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `Snapshot description.`,
+			},
+			"started": {
```
Comment (ShiChangkuo) [inline huaweicloud/services/dws/resource_huaweicloud_dws_snapshot.go]: it's better to use `created_at` or `started_at`

File: huaweicloud/services/dws/resource_huaweicloud_dws_snapshot.go [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Time when a snapshot starts to be created.`,
+			},
+			"finished": {
```
Comment (ShiChangkuo) [inline huaweicloud/services/dws/resource_huaweicloud_dws_snapshot.go]: it's better to use `finished_at`

**Pull request merged**